src/Kiota.Builder/KiotaBuilder.cs

```diff
@@ -51,6 +52,14 @@ public KiotaBuilder(ILogger<KiotaBuilder> logger, GenerationConfiguration config
         this.logger = logger;
         this.config = config;
         httpClient = client;
+        if (!Int32.TryParse(Environment.GetEnvironmentVariable("KIOTA_THREADS"), out var nrThreads))
```

Comment:
Kiota already has a configuration infrastructure. To avoid duplicating and spreading reading the configuration, can you instead update the generation configuration please?
https://github.com/microsoft/kiota/blob/b6ee9fbd7fce338875bcf09f83c8702eeafc11c9/src/Kiota.Builder/Configuration/GenerationConfiguration.cs#L10

Set the default to -1 for the time being. You can name the property something like MDOP making the Environment variable name KIOTA_GENERATION_MDOP.
Don't forget to update the app settings as well.

Reply:
I've moved the setting (an int) to the GenerationConfiguration. I'd figured the ParallelOptions was not very suitable to be placed there, so I kept that one in KiotaBuilder but made it read only and initialized from the configuration.